### PR TITLE
Chore: Add test that deprecated rules display a deprecated notice

### PR DESCRIPTION
**Pull request by wafuwafu13**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: Add tests

#### What changes did you make? (Give an overview)

I update deprecated information at https://github.com/eslint/eslint/pull/14961 and turned out we need to add some more tests.
https://github.com/eslint/eslint/pull/14961#issuecomment-903346856
>Have a This rule was deprecated... notice at the top of their rule doc
>Have `@deprecated` in their rule doc comment

### CODE REVIEW

File: docs/rules/indent-legacy.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # enforce consistent indentation (indent-legacy)
 
+This rule was **deprecated** in ESLint v4.0.0.
```
Comment (bmish) [inline docs/rules/indent-legacy.md:3]: Should have blank line after: 
```suggestion
This rule was **deprecated** in ESLint v4.0.0.

```

File: Makefile.js [hunk 2]
```diff
@@ -816,6 +816,20 @@ target.checkRuleFiles = function() {
...
+        /**
+         * Check if deprecated information is in rule code and READNE.md.
+         * @returns {boolean} true if present
+         * @private
+         */
+        function hasDeprecatedInfo() {
+            const ruleCode = cat(filename);
+            const deprecatedTagRegExp = /@deprecated in ESLint/u;
+            const docText = cat(docFilename);
+            const deprecatedInfoRegExp = /This rule was .+deprecated.+in ESLint/u;
```
Comment (bmish) [inline Makefile.js:828]: Can you specify the asterisks exactly?

```suggestion
            const deprecatedInfoRegExp = /This rule was \*\*deprecated\*\* in ESLint/u;
```

Comment (wafuwafu13) [reply Makefile.js:828]: https://github.com/eslint/eslint/blob/master/docs/rules/require-jsdoc.md and https://github.com/eslint/eslint/blob/master/docs/rules/valid-jsdoc.md are exception.

File: docs/rules/indent-legacy.md (hunk 1 above)
Comment (wafuwafu13) [reply docs/rules/indent-legacy.md:3]: Done https://github.com/eslint/eslint/pull/14989/commits/48b21c2e06d98bc8aed0a067193209ab52c760c5

**Review by bmish — APPROVED** [review]

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Review by snitin315 — APPROVED** [review]
Looks Good!

**Pull request merged**